virtual_rainforest/models/plants/plants_model.py

```diff
@@ -154,28 +151,25 @@ def from_config(
             data: A :class:`~virtual_rainforest.core.data.Data` instance.
             config: A validated Virtual Rainforest model configuration object.
             update_interval: Frequency with which all models are updated
-
         """
 
-        # Get required init arguments from config
-        soil_layers = config["core"]["layers"]["soil_layers"]
-        canopy_layers = config["core"]["layers"]["canopy_layers"]
-
         # Load in the relevant constants
         constants = load_constants(config, "plants", "PlantsConsts")
 
         # Generate the flora
         flora = Flora.from_config(config=config)
 
+        # Get the LayerStructure
+        layer_structure = LayerStructure.from_config(config=config)
+
         # Try and create the instance - safeguard against exceptions from __init__
         try:
             inst = cls(
                 data=data,
                 update_interval=update_interval,
                 flora=flora,
                 constants=constants,
-                canopy_layers=canopy_layers,
-                soil_layers=soil_layers,
+                layer_structure=layer_structure,
```

Comment:
Is this new structure going to be rolled out to all models?

Reply:
See #353! 😄